crates/agentgateway/src/http/auth.rs

```diff
@@ -292,3 +341,108 @@ mod aws {
 		}
 	}
 }
+
+mod azure {
+	use crate::{
+		client,
+		http::auth::{AzureAuth, AzureAuthCredentialSource, AzureUserAssignedIdentity},
+	};
+	use azure_core::credentials::TokenCredential;
+	use azure_identity::UserAssignedId;
+	use secrecy::ExposeSecret;
+	use std::sync::{Arc, LazyLock};
+	use tracing::trace;
+
+	static CLIENT: LazyLock<client::Client> = LazyLock::new(|| {
```

Comment:
not blocking: anywhere we have to call apply_backend_auth we have easy access to a Client we could probably re-use. In the future the others like GCP should probably use it too so it would make sense long term as well

Reply:
Good call - changed it to pass the clients through